### PR TITLE
Added angular quantities printing in latex and pretty

### LINKED ISSUE #12874: No support for \[{90^^\circ }\]
I have latex \[{90^^\circ }\], which means angle ninety degree, for example (cos(90 degree)) = 0, please add support for that? I appreate your reply.

### PR DESCRIPTION
**Pull request by Subhash-Saurabh**
Added a function `_print_Quantity` that allows latex printing of degree and radian.
```
>>> latex(90*degree)
r"90 ^\circ"
```
Also made changes so that quantities print at the end of a expression.
```
>>> expr = 2*kg*x*meter**2
>>> latex(expr, mul_symbol='dot')
2 \cdot x \cdot kilogram \cdot meter^{2}
```
#### References to other Issues or PRs
Fixes #12874

### CODE REVIEW

File: sympy/printing/latex.py
Comment (gxyd) [inline sympy/printing/latex.py]: You should add test for this case as well.

Comment (gxyd) [reply sympy/printing/latex.py]: Wait.

Comment (gxyd) [inline sympy/printing/latex.py]: No need for a test case for this line.
Just add a space between `%` and `expr`.

**Review by gxyd — COMMENTED** [review]
Also improve the pretty printing may be?

**Comment by gxyd** [comment]
You can also add latex and pretty printing for other angle units (radian, steradian, ...) as well. If you don't intend to work on that in this PR, I can open an issue to keep track of that.

**Comment by gxyd** [comment]
The wiki page for Radian mentions it using both the symbols `rad` and `^c` being in use, I'm not sure of the benefits of one compared to the other. Opinion @jksuom ? I would be more inclined towards using simply `rad`, and not `^c`, since the earlier will create no confusion.

**Comment by Subhash-Saurabh** [comment]
I am working to add degree for `prettyprinting` . I will make additional commits as soon as possible.

File: sympy/printing/latex.py
Comment (asmeurer) [inline sympy/printing/latex.py]: Use self.name. Don't rely on the str printer here.

Comment (Subhash-Saurabh) [reply sympy/printing/latex.py]: `Latexprinter` class has no attribute `name` . So using `self.name` throws `AttributeError` . Please point out if I am getting something wrong.

Comment (ylemkimon) [reply sympy/printing/latex.py]: I think he meant `expr.name`.

Comment (Subhash-Saurabh) [reply sympy/printing/latex.py]: Type of `expr.name` is Symbol. 
Therefore we have to add check like 
```
if expr.name == Symbol('degree'):
    return r"^\circ"
``` 
Is this format alright or is there something that I am missing.

Comment (ylemkimon) [reply sympy/printing/latex.py]: You can do that or you can do `expr.name.name`.

**Comment by ylemkimon** [comment]
Is this ready to merge or are you going to add other units, e.g., radian?

**Comment by Subhash-Saurabh** [comment]
I am waiting for someone to approve the changes . Once this gets approved I will add other units in the similar manner.

File: sympy/printing/latex.py
Comment (ylemkimon) [inline sympy/printing/latex.py]: `expr.args` is a tuple, which is immutable and doesn't support `del`. You have to convert it to a `list`.

**Comment by Subhash-Saurabh** [comment]
@gxyd can you please review this PR.
For steradian printing I think the empty printer is fine . What do you suggest ?

File: sympy/printing/latex.py
Comment (ylemkimon) [inline sympy/printing/latex.py]: Don't rely on the `str` printer here, too.

Comment (Subhash-Saurabh) [reply sympy/printing/latex.py]: If we use `args[i].name` instead of `str(args[i])` we get error when `args[i]` is Integer,Pow etc. Is there any other way so that we don't have to use the `str` printer and we can make this check.

**Comment by Subhash-Saurabh** [comment]
@ylemkimon I have made the changes you suggested .Can you please review it now.

**Review by ylemkimon — APPROVED** [review]

**Comment by ylemkimon** [comment]
I think this can be applied to all units.

**Comment by Subhash-Saurabh** [comment]
Current behaviour of latex and pprint
For steradian
```
>>> expr = 4*pi*steradian
>>> pprint(expr)
4⋅π⋅steradian
>>> latex(expr)
4 \\pi steradian
```
The [wiki](https://en.wikipedia.org/wiki/Steradian) mentions the above expr to be represented as `4π sr`.
This is the case for other units as well. So I think the current behaviour of `pprint` is fine.
@ylemkimon @gxyd can you please specifically mention what changes do you want in printing of other units as I am unable to figure it out.

**Comment by Subhash-Saurabh** [comment]
@ylemkimon I think for pretty printing of other units I will open a new PR also addressing the issue #13269 . So this PR can be merged maybe.

File: sympy/printing/latex.py
Comment (ylemkimon) [inline sympy/printing/latex.py]: I was thinking that printing the units at the end can be applied to all units. But if you think it's out of scope of this PR, will merge this PR.

Comment (Subhash-Saurabh) [reply sympy/printing/latex.py]: I got your point now . I will add other quantities in this PR only.

**Comment by Subhash-Saurabh** [comment]
@ylemkimon I have added for other units printing as well. Can you please take a look?

File: sympy/printing/latex.py
Comment (ylemkimon) [inline sympy/printing/latex.py]: You can do `sorted(args, key=lambda x: isinstance(x, Quantity) or (isinstance(x, Pow) and isinstance(x.base, Quantity)))`.

**Review by ylemkimon — APPROVED** [review]

File: sympy/printing/latex.py
Comment (ylemkimon) [inline sympy/printing/latex.py]: Should `rad` be raised?

**Comment by Subhash-Saurabh** [comment]
@ylemkimon The Wikipedia page of radian mentions radian to be expressed in the following form. If you want any changes kindly do suggest.

**Comment by ylemkimon** [comment]
But I'm not sure it's commonly used.

**Comment by Subhash-Saurabh** [comment]
@ylemkimon Then should we revert back to emptyprinter for radian printing or do you suggest some other changes.

**Comment by Subhash-Saurabh** [comment]
@ylemkimon I have made the suggested changes . Can you please review this PR.

**Review by ylemkimon — APPROVED** [review]

**Comment by Subhash-Saurabh** [comment]
@ylemkimon No one is suggesting any change on this PR so I think it can be merged .

File: sympy/printing/latex.py
Comment (asmeurer) [reply sympy/printing/latex.py]: If it is it needs to be `^{rad}`

**Pull request merged**
